Jeong minsoo Code review

NANNDA3463 opened the pull request:
챕터 4 코드를 추가하였습니다.

NANNDA3463 committed:
정민수 파일 업로드

NANNDA3463 committed:
정민수 파일 업로드

NANNDA3463 committed:
main rebasing

NANNDA3463 committed:
main rebasing

NANNDA3463 committed:
main rebasing

NANNDA3463 committed:
정민수 파일 업로드

NANNDA3463 committed:
Jeong minsoo file update

NANNDA3463 committed:
Jeong minsoo file update

NANNDA3463 committed:
Jeong minsoo file update

NANNDA3463 committed:
Jeong minsoo file update

NANNDA3463 committed:
Jeong minsoo file update

NANNDA3463 committed:
Jeong minsoo file update

NANNDA3463 committed:
Modify route

NANNDA3463 committed:
Modify route

NANNDA3463 committed:
Modify route

NANNDA3463 committed:
Merge remote-tracking branch 'origin/Jminsoo' into Jminsoo

# Conflicts:
#	Chapter4/pom.xml

NANNDA3463 committed:
pom modify (JavaFx)

NANNDA3463 committed:
pom modify (JavaFx)

NANNDA3463 committed:
pom modify (JavaFx)

NANNDA3463 committed:
code modify

NANNDA3463 committed:
code modify

NANNDA3463 committed:
code modify

NANNDA3463 committed:
pom modify

NANNDA3463 committed:
delete Chapter3 excercise 8

NANNDA3463 committed:
delete Chapter3 excercise 8

NANNDA3463 committed:
delete Chapter3 excercise 8

NANNDA3463 committed:
Commit for Pull Request

NANNDA3463 committed:
Commit for Pull Request

NANNDA3463 committed:
Commit for Pull Request

NANNDA3463 committed:
Merge remote-tracking branch 'origin/Jminsoo' into Jminsoo

# Conflicts:
#	Chapter3/src/main/java/com/nhnacademy/jminsoo/excercise1/Excercise3_1.java

NANNDA3463 committed:
Chapter 4 - Excercise6 Modify

NANNDA3463 committed:
Chapter 4 - Excercise6 Modify

NANNDA3463 committed:
Merge remote-tracking branch 'origin/Jminsoo' into Jminsoo

nueag commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise2/Number.java:
```diff
@@ -0,0 +1,17 @@
+        }
+
+        return Character.getNumericValue(hexNum);
+    }
```
hexValue(): 문제에는 switch문으로 구현하라고 나와있는데 확인 부탁드립니다..!

minsu11 replied:
동의합니다!

hyeonjaez replied:
동의합니당

NANNDA3463 replied:
맞습니다 수정했습니다

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise1/Excercise4_1.java:
```diff
@@ -0,0 +1,33 @@
+        Scanner scanner = new Scanner(System.in);
+        
+        String str = scanner.nextLine();
+        printCapitalized(str);
```
Stirng 객체에 데이터를 담는 것 보단 System.out.println에 바로 작성하는 건 어떨까요?? 그리고 try-catch release문을 사용하는 건 어떨까요?

prospro1 replied:
저는 오히려 여기서 str이라는 변수를 생성해서 받는 게 더 코드를 읽기에 분명하고 좋지 않을까 생각합니다...! 혹시 다른 분들은 두 방법 중 어떤 게 더 좋다고 생각하시는지 의견을 나누고 싶습니다! 이걸 정해야지 앞으로 코드에 대한 규칙들이 쌓여 비슷하게 코드를 짤 수 있게 될 것 같아요!

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise3/Main.java:
```diff
@@ -0,0 +1,64 @@
+        catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+        }
+        finally {
```
try-catch release문을 작성하는 건 어떨까요??

NANNDA3463 replied:
좋습니다 수정하였습니다

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise3/Main.java:
```diff
@@ -0,0 +1,64 @@
+import java.util.InputMismatchException;
+import java.util.Scanner;
+
+public class Main {
```
4.3문제가 주사위 뱀눈 판별이여서 number로 2가 아닌 다른 숫자가 들어오면 실행이 안되는 방향으로 했으면 어땠을 까요?

nueag commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/DiceApp.java:
```diff
@@ -0,0 +1,47 @@
+    private final int diceCount;
+    private final int eyes;
+
+    private final Dice[] dice;
```
dice배열이기 때문에 dices로 하면 어떨까요?

NANNDA3463 replied:
놀랍게도 dice가 복수형이라고 합니다. 단수형은 die라고 하네요
헉!

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,25 @@
+    public final static int DICE_COUNT = 2;
+    public final static int EYES = 6;
+    public static void main(String[] args) {
+        Dice[] dice = new Dice[DICE_COUNT];
```
main에서 생성하는게 아니라 DiceApp에 Dice를 생성을 하는 방식을 어떨까요?

prospro1 replied:
동의합니다 :)

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise6/RandomMosaicWalk2.java:
```diff
@@ -0,0 +1,119 @@
+public class RandomMosaicWalk2 {
+
+    final static int ROWS = 20;        // Number of rows in mosaic.
+    final static int COLUMNS = 20;     // Number of columns in mosaic.
```
문제에서 말하는 조건과 다른거 같습니다!

NANNDA3463 replied:
빨리 결과를 보고 싶어 줄였습니다 미안합니다

nueag commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,25 @@
+        System.out.println("주사위의 총합 \t 평균 굴림 횟수");
+        System.out.println("----------\t----------");
+
+        for (int i = DICE_COUNT; i <= DICE_COUNT * EYES; i++) {
```
확장성이 좋은 것 같습니다!b

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise6/RandomMosaicWalk2.java:
```diff
@@ -0,0 +1,119 @@
+     */
+    public static void main(String[] args) {
+        Mosaic.open(ROWS, COLUMNS, SQUARE_SIZE, SQUARE_SIZE);
+        fillWithRandomColors();
```
검정색으로 칠하는 fillWithBlackColors를 사용을 안했습니다!

prospro1 replied:
동의합니다!

NANNDA3463 replied:
감사합니다 수정하였습니다

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise7/RandomMosaicWalk2.java:
```diff
@@ -0,0 +1,152 @@
+    static void fillGreenColors(int rowNum, int colNum ) {
+        Mosaic.setColor(rowNum, colNum, 0, Mosaic.getGreen(rowNum, colNum) + 25, 0);
+    }
+    static void fillWithBlackColors() {
```
사용하지 않는 메소드는 삭제해도 될 것 같습니다!

NANNDA3463 replied:
동의합니다 삭제하였습니다

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+            try {
+                answer = scanner.nextInt();
+            } catch (InputMismatchException e) {
+                scanner.close();
```
for문 위에 try-catch release문을 사용하면 좋을 것같습니다!!

NANNDA3463 replied:
좋은 리뷰 감사합니다

minsu11 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+    public void quizGenerate() {
+        for (int i = 0 ; i < quizzes.length; i++) {
+            int firstNum = (int)(Math.random()*100);
+            int secondNum = (int) (Math.random()*100);
```
100을 넣는 것보단 변수에 값에 100을 넣어서 사용하면 괜찮을 것  같습니다!

NANNDA3463 replied:
수정하였습니다

minsu11 reviewed (commented):
parkminsu code review

hyeonjaez commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/DiceApp.java:
```diff
@@ -0,0 +1,47 @@
+        this.dice = dice;
+    }
+
+    public int correctCount(int number) {
```
메서드 이름 첫 단어는 동사로 하면 좋을거 같아요 예를들어서 countCorrectNumber 이런 식으로 하면 더 명확해서 좋을거 같아요

NANNDA3463 replied:
동의합니다 둘의 순서를 반대로 countCorrect로 하였습니다.

hyeonjaez commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/DiceApp.java:
```diff
@@ -0,0 +1,47 @@
+        return count;
+    }
+
+    public double getDiceAverage(int times, int number) {
```
변수명 get 보단 계산을 하는 메서드이기 때문에 calculateDiceAverage가 더 맞는 뜻 같아요

NANNDA3463 replied:
맞습니다 감사합니다

hyeonjaez commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,35 @@
+        };
+    }
+
+    public static ArrayProcessor sum = (x) -> Arrays.stream(x).sum();
```
오옹 stream 보고 배울게요

hyeonjaez commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+    @Override
+    public void quizGenerate() {
+        for (int i = 0 ; i < quizzes.length; i++) {
+            int firstNum = (int)(Math.random()*100);
```
정렬 한번 해주세요 참고로 단축키는 옵션 + 커맨드+ L 입니다 ㅎ

NANNDA3463 replied:
코드 스타일에 맞게 수정하겠습니다

hyeonjaez reviewed (commented):
오늘 stream 배운거 바로 응용하는거 좋았습니다

NANNDA3463 committed:
excercise 5-1 commit

NANNDA3463 committed:
Chapter 5 Upload

NANNDA3463 committed:
RandomMosaicWalk2 modify

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise1/DiceApp.java:
```diff
@@ -0,0 +1,58 @@
+        return count;
+    }
+
+    public double calculateDiceAverage(int times, int number) {
```
사용하지 않는 메소드이라면 삭제하는 게 좋을 것같습니다!!

NANNDA3463 replied:
삭제하였습니다

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,105 @@
+
+        System.out.println();
+        System.out.println("게임 횟수 :  " + gamesPlayed + " games.");
+        System.out.printf("평균 점수 : %1.3f.\n", averageScore);
```
실행 했을 때 평균 점수가  나오지 않습니다! 수정 부탁드립니다!

NANNDA3463 replied:
이긴 횟수 / 게임 횟수인데 이기기가 좀 어려워서 0을 벗어나기가 힘들긴 합니다

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+            bet = TextIO.getlnInt();
+            if (bet > money) {
+                System.out.printf("베팅 금액은 현재 금액보다 많을 수 없습니다. 현재 돈 : %d\n", money);
+                continue;
```
처음 실행했을 때 100보다 크게 입력 했을 경우 프로그램에서 다시 작동이 되게 만들면 좋을 것 같습니다! 지금은 150을 넣었을 때 다시 입력을 받는 것이 아니라 프로그램이 멈췄습니다!!

NANNDA3463 replied:
조정하였습니다

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+        } while (playAgain);
+
+        averageScore = ((double) sumOfScores) / gamesPlayed;
+
```
게임을 끝내고나서 플레이어의 베팅금액도 출력하게하면 좋을 것 같습니다!

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+                    System.out.print("새 카드를 받으시겠습니까? (Stand / Heat) : ");
+                    answer = TextIO.getln();
+
+                    if (answer.equals("Stand") || answer.equals("Heat")) {
```
소문자로 stand 또는 heat라고 적었을 때도 작동이 되게 하면 좋을거 같습니다!!

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+        }
+
+        //둘 다 21일 경우
+        if (player.getHandValue() == 21 && dealer.getHandValue() == 21) {
```
딜러가 21이 되면 무조건 플레이어가 지는 조건이기 때문에 해당 조건문은 제외해도 괜찮지 않을까요? 밑에 있는 조건문이 해당 조건문을 대체할 수 있을 것 같아요!

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+                } while (!answer.equals("Stand") && !answer.equals("Heat"));
+
+                if (answer.equals("Stand")) {
+                    if (dealer.getHandValue() < 16) {
```
딜러의 패가 16이하 일 경우에 패를 받기때문에 16을 포함이 되어야합니다!!

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+        //둘 다 21일 경우
+        if (player.getHandValue() == 21 && dealer.getHandValue() == 21) {
+            return false;
+        } else if (dealer.getHandValue() == 21) {
```
play메소드 안에 조건문과 비슷한 구조를 가진 조건문인거 같아서 메소드로 따로 뺴서 만드는 게 좋을 것 같습니다!

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+            } else {
+                do {
+                    System.out.print("새 카드를 받으시겠습니까? (Stand / Heat) : ");
+                    answer = TextIO.getln();
```
answer이라는 변수 이름이 해당 프로그램과 어울리지 않은 것 같습니다! 수정이 가능하실까요??

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise6/Quiz.java:
```diff
@@ -0,0 +1,59 @@
+
+        for (int i = 0; i < quizzes.length; i++) {
+            System.out.println(quizzes[i].getQuestion());
+            try {
```
try -catch release문을 사용하시면 어떨까요??

NANNDA3463 replied:
수정하였습니다

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise7/Main.java:
```diff
@@ -0,0 +1,11 @@
+package com.nhnacademy.Jminsoo.excercise7;
+
+public class Main {
+    public static void main(String[] args) {
```
프로그램이 실행이 되지 않습니다! ArrayStoreException이 뜹니다!!

minsu11 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise7/Quiz.java:
```diff
@@ -0,0 +1,74 @@
+        for (int i = 0; i < quizzes.length / 2 - 1; i++) {
+            quizzes[i] = new AdditionQuestion();
+        }
+        for (int i = 0; i < quizzes.length / 2; i++) {
```
loop문의 조건식이 잘못되어있어서 작동이 안되는것 같습니다! i = 0이 아니라 quizzes.length / 2부터 시작되어야 할 것같습니다!!

NANNDA3463 replied:
조건식도 문제가 있었고, Constructor에서 quizzes에다 AdditionQuestion을 넣은 것도 컸습니다.
조건식은 맞게 for문을 고치고,  quizzes에 IntQuestion을 넣었습니다.
둘 다 수정하였습니다.

minsu11 reviewed (commented):
pakrminsu code review

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise2/Main.java:
```diff
@@ -0,0 +1,26 @@
+        Scanner scanner = new Scanner(System.in);
+        System.out.print("16진수 값을 입력해주세요 : ");
+
+        String hexNum = scanner.next();
```
hexNum 이라고 변수명 지으신 거 되게 좋은 것 같아요!!

NANNDA3463 replied:
감사합니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise2/Number.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    public static boolean isHex(char c) {
+        int numericValue =Character.getNumericValue(c);
```
사실 여기서 Character.getNumericValue(c)로 받아서 체크를 해줄 때, 16진수가 아닌 값을 보냈는지 확인할 수도 있었을 것 같아요!

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise2/Number.java:
```diff
@@ -0,0 +1,17 @@
+            throw new IllegalArgumentException("16진수가 아닌 값을 보냈습니다!");
+        }
+
+        return Character.getNumericValue(hexNum);
```
여기서 보시면 Character.getNumericValue() 메서드가 여기서도 사용되고, 아래의 isHex() 에서도 사용되므로, 두 번 값을 변환할 필요는 없지 않을까 싶습니다!

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise2/Number.java:
```diff
@@ -0,0 +1,17 @@
+    }
+
+    public static boolean isHex(char c) {
+        int numericValue =Character.getNumericValue(c);
```
Character.getNumericValue(c) 가 여기 isHex() 에서 먼저 사용이 되고, 위의 hexValue() 에서도 리턴할 때도 사용이 되고 있습니다..! 중복되서 사용되는 걸 조금 개선을 할 수 있으면 좋지 않을까 생각해봅니다...!

NANNDA3463 replied:
Number class를 static으로 사용하고 있었지만, Dynamic Method로 바꾸고 필드로 변수를 선언하여 사용하면 가능할 것 같습니다.
답변 감사합니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise3/Main.java:
```diff
@@ -0,0 +1,64 @@
+        }
+        catch (IllegalArgumentException e) {
+            System.out.println(e.getMessage());
+        }
```
catch 들의 위치가 잘 정렬되게, 띄어쓰기로 해주시면 보기가 더 좋을 것 같습니다 ㅎㅎ

NANNDA3463 replied:
맞습니다 코드 스타일 대로 수정하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise3/Main.java:
```diff
@@ -0,0 +1,64 @@
+        }
+    }
+
+    private static int correctCount(int number, Dice[] dice) {
```
메서드 명을 correctCount 라고 작성하신 이유가 궁금합니다...!

NANNDA3463 replied:
correct 할 때까지의 Count라고 짧게 생각했습니다.

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise2/Main.java:
```diff
@@ -0,0 +1,35 @@
+        } catch (NoSuchElementException | IllegalStateException e) {
+            System.out.println("잘못된 값이 들어갔습니다 ! 오류 메세지 : " + e.getMessage());
+        }
+        System.out.printf("넣은 개수 : %d\n", calc.getCount());
```
StatCalc 안에 toString 을 오버라이딩 하면 어떨까요??

NANNDA3463 replied:
동의합니다 반영하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/DiceApp.java:
```diff
@@ -0,0 +1,47 @@
+        while(!isCorrect) {
+            eyesSum = 0;
+            count++;
+            for (Dice die : dice) {
```
위에서 가은씨가 제안하신 것처럼, dices로 변경한 후, 여기서 die 대신 dice로 적을 수 있으면 더 좋을 것 같아요!

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise3/Dice.java:
```diff
@@ -0,0 +1,24 @@
+        this.diceRoll();
+    }
+
+    public void diceRoll() {
```
액션이니까 동사가 앞으로 오면 더 좋지 않을까 생각해봅니다...!

NANNDA3463 replied:
동의합니다 감사합니다

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,105 @@
+        System.out.printf("평균 점수 : %1.3f.\n", averageScore);
+    }
+
+    private static boolean play() {
```
이 메서드가 하는 행동이 많은 거 같습니다 조금씩 나누어 주는게 어떨까요?

NANNDA3463 replied:
패를 보여주는 show와 heat와 stand를 구분하는 메서드들로 나누었습니다

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,105 @@
+    }
+
+    private static boolean play() {
+        final int blackJackMaxNum = 21;
```
상수는 대문자로 이름을 지어줘야 합니다!

NANNDA3463 replied:
대문자로 변경하고 위치도 클래스 상부로 조정하였습니다

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise6/Quiz.java:
```diff
@@ -0,0 +1,59 @@
+    private int[] answer;
+
+    public Quiz(int count) {
+        quizzes = new AdditionQuestion[count];
```
this 키워드를 붙여주는게 어떨까요?

NANNDA3463 replied:
맞습니다 반영하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/DiceApp.java:
```diff
@@ -0,0 +1,47 @@
+    public double getDiceAverage(int times, int number) {
+        int countSum = 0;
+        for (int i = 0; i < times; i++) {
+            countSum += correctCount(number);
```
countSum 이라는 변수명이 마치 합계를 센다 라고 느껴지는데, 사용하신 의도는 그게 아니라 롤 카운트들의 합계  같네요!

NANNDA3463 replied:
넵 둘의 순서를 바꿨습니다. 감사합니다

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise7/Main.java:
```diff
@@ -0,0 +1,11 @@
+package com.nhnacademy.Jminsoo.excercise7;
+
+public class Main {
+    public static void main(String[] args) {
```
프로그램 실행 즉시 Exception이 일어납니다 확인 부탁드립니다

NANNDA3463 replied:
확인했습니다 수정하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/DiceApp.java:
```diff
@@ -0,0 +1,47 @@
+    public double getDiceAverage(int times, int number) {
+        int countSum = 0;
+        for (int i = 0; i < times; i++) {
+            countSum += correctCount(number);
```
그래서 저는 다들 변수명 적는 방식이 다 다르기 때문에, 변수명을 어떻게 적으면 좋을지에 대해 논의하는 시간을 가졌으면 합니다 ㅎㅎ

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise7/Quiz.java:
```diff
@@ -0,0 +1,74 @@
+
+    @Override
+    public void quizGenerate() {
+        for (int i = 0; i < quizzes.length / 2 - 1; i++) {
```
위에 작성하신 for문 조건식은 맞지 않아서 exception이 일어난거 같아요
덧셈 문제와 뺄셈문제를 번갈아 가면서 만들고 싶으시면 i가 짝수일때 와 i가 홀수 일때를 for문에 조건식을 알맞게 추가하셔서 하면 될거 같아요

NANNDA3463 replied:
조건식도 문제가 있었고, Constructor에서 quizzes에다 AdditionQuestion을 넣은 것도 컸습니다. 
조건식은 맞게 for문을 고치고,  quizzes에 IntQuestion을 넣었습니다.
둘 다 수정하였습니다.

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise7/Quiz.java:
```diff
@@ -0,0 +1,74 @@
+    private int[] answer;
+
+    public Quiz(int count) {
+        quizzes = new AdditionQuestion[count];
```
this 키워드를 붙여 주는게 좋을거 같아요

NANNDA3463 replied:
맞습니 다 수정하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/DiceApp.java:
```diff
@@ -0,0 +1,47 @@
+
+    private final Dice[] dice;
+
+    public DiceApp(int diceCount, int eyes, Dice[] dice) {
```
여기 DiceApp 생성자에서 파라미터로 받은 Dice[] dice 를   위에 field로 선언해두신 Dice[] dice 에 "this.dice = dice" 이렇게 할당해서 초기화해주시면 좋을 것 같아요 :)

NANNDA3463 replied:
감사합니다 수정하였습니다.

nueag commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+                } else if (answer.equals("Heat")) {
+                    card = deck.dealCard();
+                    System.out.println("\n\n다음 카드 : " + card);
+                    player.getHand().addCard(card);
```
사용자의 카드가 추가된 이후에 카드의 합이 21이 되는지 확인하는 부분이 없는 것 같습니다..!

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+                money += bet;
+                sumOfScores++;
+            } else {
+                System.out.printf("베팅 금액을 잃습니다. -%d\n", bet);
```
배팅 금액을 잃었고 그뒤에 현재 돈이 얼마가 남았는지도 알려주면 좋을거 같아요

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,26 @@
+
+        for (int i = 0; i < DICE_COUNT; i++) {
+            dice[i] = new Dice(EYES);
+        }
```
이 위의 for 문도 Dice 배열에 Dice를 생성해서 넣어주는 거니까 Dice 배열을 field로 사용하고 있는 DiceApp 클래스의 생성자 부분에 넣어주면 좋을 것 같아요!

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise5/Main.java:
```diff
@@ -0,0 +1,167 @@
+            boolean isVictory;        // Score for one game.
+
+            //돈 걸기
+            System.out.println("베팅 금액을 설정해주세요");
```
배팅 금액 0원 일때도 다시 예외처리를 해주는게 좋을거 같아요

NANNDA3463 replied:
0이하로 걸고 지면 돈이 복사될 뻔 했습니다 수정하였습니다

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,105 @@
+
+        deck.shuffle();
+
+        System.out.print("받고 싶은 카드의 개수를 입력 하세요 (2 ~ 6) : ");
```
2~6 사이의 숫자가 아니라면 다시 입력하게 만들어주면 좋을거 같아요

hyeonjaez reviewed (commented):
comment

prospro1 reviewed (commented):
yunhwa reviewed up to Chapter4 - exercise4 of Jeong minsoo

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise6/RandomMosaicWalk2.java:
```diff
@@ -0,0 +1,119 @@
+    /**
+     * 추가한 메서드
+     */
+    static void fillGreenColors(int rowNum, int colNum) {
```
green 요소값이 255가 초과될 때도 계속 색상을 더해주려고 시도할 것 같습니다! 255 이하일 때만 돌아가게 조건을 걸어주시면 불필요한 동작을 더이상 진행시키지 않을 것 같습니다 :)

NANNDA3463 replied:
좋은 지적 감사합니다 수정하였습니다.

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise7/RandomMosaicWalk2.java:
```diff
@@ -0,0 +1,152 @@
+        fillWithRandomColors();
+
+        currentRow = ROWS / 2;   // start at center of window
+        currentColumn = COLUMNS / 2;
```
문제의 조건에서 처음에 모자이크의 직사각형 중 하나를 무작위로 선택한다 하였으므로, 중앙의 점이 아니라 랜덤으로 설정해주시면 좋을 것 같습니다!

NANNDA3463 replied:
문제를 잘못읽었습니다 수정하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise7/RandomMosaicWalk2.java:
```diff
@@ -0,0 +1,152 @@
+
+    static void changeToCurrentColor(int rowNum, int colNum) {
+
+        int adjacentNum = (int)(4*Math.random());
```
변수명에 인접하다는 표현을 사용하신 거 너무 좋은 것 같아요!

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise7/RandomMosaicWalk2.java:
```diff
@@ -0,0 +1,152 @@
+                tempCol--;
+                if (tempCol < 0)
+                    tempCol = 0;
+            }
```
case 문에서 '->' 로 설정해주지 않고 이렇게 콜론(:) 으로 작성해도, 중괄호{} 를 해주면, break 문 작성 안해주어도 case 0 일 때, case 3 부분까지 다 거치고 오진 않는지 궁금합니다! 설명해주시면 감사하겠습니다 :)

NANNDA3463 replied:
이해하고 계신 바가 정확합니다.  break를 깜빡했습니다.

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+            int sum = quiz.getFirstNum() + quiz.getSecondNum();
+            System.out.printf("정답 : %d + %d = %d, 입력값 : %d, 정답 유무 : %b\n", quiz.getFirstNum(), quiz.getSecondNum(), sum, quiz.getAnswerNum(),sum ==
+                    quiz.getAnswerNum() );
+            if (sum == quiz.getAnswerNum()) {
```
위의 System.out.printf 문에서도 동일하게 같은지를 확인하고 있습니다! 이런 경우에는 boolean 변수를 하나 생성해서 값을 저장해두고 사용하는 건 어떨까요~?

NANNDA3463 replied:
동의합니다 반영했습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/SkeletonQuiz.java:
```diff
@@ -0,0 +1,7 @@
+    void quizGenerate();
+    void quizControl();
+    void quizGrading();
+}
```
인터페이스로 작성하신 거 좋은 것 같습니다!!

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+public class AddtionQuiz implements SkeletonQuiz {
+    private Quiz[] quizzes;
+    private double correct;
+
```
문제의 조건에서는 첫 번째 값에 대한 배열, 두 번째 값에 대한 배열, 그리고 사용자의 답변에 대한 배열, 이렇게 배열 3개를 각각 사용하라고 되어있었지만.. 이렇게 quiz가 여러 개 담긴 배열을 생성하는 접근 방식은 참 좋은 것 같습니다!

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+
+public class AddtionQuiz implements SkeletonQuiz {
+    private Quiz[] quizzes;
+    private double correct;
```
문제의 조건에서는 첫 번째 값에 대한 배열, 두 번째 값에 대한 배열, 그리고 사용자의 답변에 대한 배열, 이렇게 배열 3개를 각각 사용하라고 되어있었지만.. 이렇게 quiz가 여러 개 담긴 배열을 생성하는 접근 방식은 참 좋은 것 같습니다!

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+
+public class AddtionQuiz implements SkeletonQuiz {
+    private Quiz[] quizzes;
+    private double correct;
```
그리고 correct는 사용자가 맞춘 문제 개수인 것 같습니다! 그렇다면 correctCount 이런 식으로 개수라는 게 느껴지게 변수명을 변경해보시는 건 어떨까요~?

NANNDA3463 replied:
맞습니다 변경하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/AddtionQuiz.java:
```diff
@@ -0,0 +1,57 @@
+            }
+        }
+
+        System.out.printf("최종 점수 : %.2f %%", (correct / quizzes.length) * 100.0);
```
'%' 로 표기해주시고 싶은 거라면 '점수' 보다는 '정답률' 과 같은 표현이 보다 더 알맞을 것 같습니다!!

NANNDA3463 replied:
맞습니다 수정하였습니다

prospro1 commented on Chapter4/src/main/java/com/nhnacademy/jminsoo/excercise8/Quiz.java:
```diff
@@ -0,0 +1,26 @@
+    public int getAnswerNum() {
+        return problemArr[2];
+    }
+}
```
퀴즈 클래스를 만드신 거 되게 좋은 것 같습니다!!

prospro1 reviewed (commented):
yunhwa reviewed up to Chapter4 last exercise!

NANNDA3463 committed:
Chapter 4 Modify (리뷰 반영)

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise1/DiceApp.java:
```diff
@@ -0,0 +1,58 @@
+        DiceGenerater(diceCount, eyes);
+    }
+
+    private void DiceGenerater(int diceCount, int eyes) {
```
메서드 명을 DicesGenerater 로 변경해주시면 여러 개의 주사위들이 생성되는 이 메서드 동작의 의미에 보다 걸맞을 것 같습니다 :) 참고 부탁드립니다!

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise1/DiceApp.java:
```diff
@@ -0,0 +1,58 @@
+        for (int i = 0; i < diceCount; i++) {
+            dices[i] = new Dice(eyes);
+        }
+        this.dices = dices;
```
그리고 Dice[] 을 하나 더 생성해서 다 채워진 걸 field에 대입하는 것보다는, 처음부터 DiceApp 에 field 로 선언되어 있는 dices 에 바로 Dice 생성자로 할당해주는 게 더 좋지 않을까 싶습니다!

prospro1 replied:
이 메서드는 결국 DiceApp 생성자 안에서만 사용되는 것으로 보이기 때문에 이렇게 추천드려봅니다 :)

NANNDA3463 committed:
Chapter 5 Modify (리뷰 반영)

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise1/Dice.java:
```diff
@@ -0,0 +1,24 @@
+    public int getDiceNum() {
+        return diceNum;
+    }
+}
```
해당 클래스를 작성하신 것은 너무 좋습니다! 다만, 1번 문제의 요구사항을 맞춰서 프로그램을 짜보시는 것도 해보시면 어떨까요~?

- 문제의 요구사항은 아래와 같습니다
1. 객체 field인 die1 및 die2 를 private으로 바꾼, 다른 버전의 pairOfDice 클래스를 작성
2. 두 주사위의 합이 2가 될 때까지 주사위 한 쌍을 굴린 횟수를 계산하는 짧은 프로그램으로 해당 클래스를 테스트

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise1/Main.java:
```diff
@@ -0,0 +1,25 @@
+        try (Scanner scanner = new Scanner(System.in)) {
+            System.out.print("주사위의 합계를 입력해주세요 : ");
+            int number = scanner.nextInt();
+            System.out.printf("모두 %d 이(가) 나올 때까지 주사위를 굴린 횟수 : %d\n", number, diceApp.countCorrectNumber(number));
```
Dice 클래스에서 코멘트 남긴 것에서도 보시다시피, 문제에서 제시한 프로그램은 주사위의 합계가 2일 때, 즉, 뱀의 눈이 될 때까지의 값을 계산하는 거라서, 입력값을 2로만 넣어주셔야할 것 같습니다!

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise1/Main.java:
```diff
@@ -0,0 +1,25 @@
+        DiceApp diceApp = new DiceApp(DICE_COUNT, EYES);
+
+        try (Scanner scanner = new Scanner(System.in)) {
+            System.out.print("주사위의 합계를 입력해주세요 : ");
```
Dice 클래스에서 코멘트 남긴 것에서도 보시다시피, 문제에서 제시한 프로그램은 주사위의 합계가 2일 때, 즉, 뱀의 눈이 될 때까지의 값을 계산하는 거라서, 입력값을 2로만 넣어주셔야할 것 같습니다!

NANNDA3463 replied:
반영하였습니다

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise2/Main.java:
```diff
@@ -0,0 +1,30 @@
+                if (inputNum == 0) {
+                    break;
+                }
+                calc.enter(inputNum);
```
enter() 메서드에 double 값이 파라미터로 들어가는데, nextInt()로 받으셔서 그런지 실제로 double 값을 "1.1" 처럼 넣으면 잘못된 값이 들어갔다는 메세지가 뜨고 종료가 되어버리는 것 같아요! 이걸 double 값도 입력이 되게 해주시면 좋을 것 같습니다 :)

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise3/Main.java:
```diff
@@ -0,0 +1,37 @@
+                    i,
+                    statCalc.getMean(),
+                    statCalc.getStandardDeviation(),
+                    statCalc.getMax());
```
출력 형식이 너무 깔끔하고 좋습니다!! 한 수 배워갑니다 ㅎㅎ

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise3/Dice.java:
```diff
@@ -0,0 +1,27 @@
+    public int getDiceNum() {
+        return diceNum;
+    }
+}
```
이 문제 또한 문제에서 요구했던, pairOfDice 를 활용하지 않은 게 아쉽긴 합니다...! 시간 나시면 한 번 해보심이 어떠신지요 :)

prospro1 reviewed (commented):
yunhwa reviewed up to Chapter5 - exercise4

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,105 @@
+        System.out.printf("현재 카드 장수 : %d\n", hand.getCardCount());
+    }
+
+    private static char heatOrStand() {
```
heat 을 hit 으로 변경해주시면 좋을 것 같습니다!!

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise4/Main.java:
```diff
@@ -0,0 +1,105 @@
+                System.out.printf("받은 카드가 %d보다 큽니다. 패배합니다.\n", blackJackMaxNum);
+                return false;
+            } else if (hand.getBlackjackValue() == blackJackMaxNum) {
+                System.out.printf("%d와 값이 맞습니다! 승립합니다!", blackJackMaxNum);
```
승리합니다! 로 오타 고쳐주시면 좋겠습니다!

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise6/Quiz.java:
```diff
@@ -0,0 +1,56 @@
+    private AdditionQuestion[] quizzes;
+    private double correct;
+
+    private int[] answer;
```
answers 처럼 여러 개가 들어간 배열이라는 점이 드러나게 변수명을 변경해주심이 어떠신지요 :)

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise6/SkeletonQuiz.java:
```diff
@@ -0,0 +1,9 @@
+    void quizControl();
+
+    void quizGrading();
+}
```
interface 로 만드신 거 너무 좋은 것 같습니다!!

prospro1 commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise7/Quiz.java:
```diff
@@ -0,0 +1,74 @@
+            @Override
+            public String getQuestion() {
+                return "만든 사람의 나이는 몇살???";
+            }
```
너무 재미있는 퀴즈인 것 같습니다!!

NANNDA3463 committed:
FIX (excercise5_1) : SNAKE_EYES를 의미하는 상수 추가. 출력 문 toString으로 변경

prospro1 reviewed (commented):
yunhwa reviewed up to all the exercises of Chapter5

NANNDA3463 committed:
FIX (excercise5_2) : 입력으로 받는 자료형을 int -> double로 수정

NANNDA3463 committed:
RENAME (Chapter4, 5) : main클래스의 이름 수정

NANNDA3463 committed:
STYLE(Excercise5_3) : import 경로 수정

hyeonjaez commented on Chapter5/src/main/java/com/nhnacademy/Jminsoo/excercise4/Excercise5_4.java:
```diff
@@ -0,0 +1,105 @@
+public class Excercise5_4 {
+    public static void main(String[] args) {
+
+        int gamesPlayed = 0;
```
게임 진행한 횟수를 세는 변수인거 같은데 이름을 명확하게 countGamePlayed 이런식으로 만들어 주는건 어떨까요?

NANNDA3463 replied:
수정하였습니다

NANNDA3463 committed:
STYLE(Excercise5_4) : 8번줄, gamePlayed 변수명을 countGamePlayed로 변경

NANNDA3463 committed:
FEAT Chapter6

NANNDA3463 committed:
FEAT Chapter7 add

NANNDA3463 committed:
REFACTOR : 4_6 수정하였습니다

NANNDA3463 committed:
FEAT(Main-Chapter4) Excercise4_1, Excercise4_6 추가

NANNDA3463 committed:
FEAT(Main-Chapter5)
- Excercise5_1 : RENAME class name : MakeSnakeEye -> SnakeEye
- Excercise5_6 : FIX bug :
try-catch-resource로 Scanner를 사용할 때, 한번 반복하고 종료하기 때문에 적합하지 않음

minsu11 reviewed (approved):
code approve

hyeonjaez reviewed (approved):
code approve

nueag reviewed (approved):
code approve

NANNDA3463 committed:
FEAT(Chapter7_3, Chapter7_5)
- Excercise 구현

NANNDA3463 committed:
Merge branch 'main' into Jminsoo
